pkg/admissioncontroller/webhook/admission/admissionpluginsecret/handler.go

```diff
@@ -0,0 +1,182 @@
+	shoot := &gardencore.Shoot{}
+	if err := runtime.DecodeInto(internalDecoder, request.Object.Raw, shoot); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	if shoot.DeletionTimestamp != nil {
+		// don't validate shoot if it's already marked for deletion
+		return admissionwebhook.Allowed("shoot is already marked for deletion")
+	}
+
+	if request.Operation == admissionv1.Update {
+		oldShoot := &gardencore.Shoot{}
+		if err := runtime.DecodeInto(internalDecoder, request.OldObject.Raw, oldShoot); err != nil {
+			return admission.Errored(http.StatusInternalServerError, err)
+		}
+
+		// skip verification if spec wasn't changed
+		// this way we make sure, that users/gardenlet can always annotate/label the shoot if the spec doesn't change
+		if apiequality.Semantic.DeepEqual(oldShoot.Spec, shoot.Spec) {
+			return admissionwebhook.Allowed("shoot spec was not changed")
+		}
+	}
+
+	if err := h.validateAdmissionPluginSecret(ctx, shoot); err != nil {
+		return admission.Denied(err.Error())
+	}
+
+	return admissionwebhook.Allowed("all referenced admission plugins secret is valid")
+}
+
+func (h *Handler) admitSecret(ctx context.Context, request admission.Request) admission.Response {
+	var secret = &corev1.Secret{}
+
+	if request.Operation != admissionv1.Update {
+		return admissionwebhook.Allowed("operation is not update")
+	}
+
+	if err := h.decoder.Decode(request, secret); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	// lookup if secret is referenced by any shoot in the same namespace
+	shootList := &gardencorev1beta1.ShootList{}
+	if err := h.Client.List(ctx, shootList, client.InNamespace(request.Namespace)); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	for _, shoot := range shootList.Items {
+		if shoot.Spec.Kubernetes.KubeAPIServer != nil {
+			for _, plugin := range shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins {
+				if plugin.KubeconfigSecretName != nil && *plugin.KubeconfigSecretName == request.Name {
+					_, ok := secret.Data[kubernetes.KubeConfig]
+					if !ok {
+						return admission.Denied(fmt.Sprintf("data kubeconfig can't be removed from secret because secret is in use by shoot: %s", shoot.Name))
+					}
+				}
+			}
+		}
+	}
+
+	return admissionwebhook.Allowed("secret change is valid")
+}
+
+func (h *Handler) validateAdmissionPluginSecret(ctx context.Context, shoot *gardencore.Shoot) error {
+	fldPath := field.NewPath("spec", "kubernetes", "kubeAPIServer", "admissionPlugins")
+
+	if shoot.Spec.Kubernetes.KubeAPIServer != nil {
+		for i, admissionPlugin := range shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins {
+			if admissionPlugin.KubeconfigSecretName != nil {
+				if err := h.validateReferencedSecret(ctx, *admissionPlugin.KubeconfigSecretName, shoot.Namespace, fldPath.Index(i).Child("kubeconfigSecretName")); err != nil {
+					return err
+				}
+			}
```

Comment:
You could also validate all of them and aggregate potential errors instead of exiting after the first failure